### PR TITLE
[Kubernetes] enable Kueue priority-class label from {CLI, SDK, YAML}

### PR DESCRIPTION
**Pull request by luccabb**
`resources.priority_class` (settable via `sky launch --priority <name>`, added in #9794) is currently only recorded as a `skypilot-priority-class` pod annotation, which Kueue does not read — so the flag has no scheduling effect on Kueue-managed clusters, this PR enables it for Kueue.

Tested (run the relevant ones):

- [x] Code formatting: install pre-commit (auto-check on commit) or `bash format.sh` (yapf + isort clean on changed files)
- [x] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `/smoke-test` (CI) or `pytest tests/test_smoke.py` (local)
- [ ] Relevant individual tests: `/smoke-test -k test_name` (CI) or `pytest tests/test_smoke.py::test_name` (local)
- [ ] Backward compatibility: `/quicktest-core` (CI) or `pytest tests/smoke_tests/test_backward_compat.py` (local)

**Manual test** — local API server from this branch + a k3d cluster (Kueue controller not required to verify labeling), with config:

```yaml
kubernetes:
  kueue:
    local_queue_name: smoke-queue
```

```console
$ sky launch -y -c pri-smoke --infra k8s/k3d-ac2 --cpus 0.5+ --priority high-priority -- 'echo hi'
$ kubectl get pod pri-smoke-f736945b-head -o jsonpath='{.metadata.labels}'
... "kueue.x-k8s.io/priority-class":"high-priority","kueue.x-k8s.io/queue-name":"smoke-queue" ...
```

### CODE REVIEW

File: sky/templates/kubernetes-ray.yml.j2 [hunk 1]
```diff
@@ -300,6 +300,9 @@ available_node_types:
             {% if k8s_kueue_local_queue_name %}
             kueue.x-k8s.io/queue-name: {{k8s_kueue_local_queue_name}}
             kueue.x-k8s.io/pod-group-name: {{cluster_name_on_cloud}}
+            {% if priority_class is not none %}
+            kueue.x-k8s.io/priority-class: {{priority_class}}
```
[the remark this answers is not in the document]
Comment (luccabb) [reply sky/templates/kubernetes-ray.yml.j2]: Good catch — done in e832338. Verified the edge case locally: with `|tojson`, class names like `"100"` and `"true"` parse back as YAML strings; unquoted, `100` parses as an int as you noted.

File: tests/unit_tests/test_backend_utils.py [hunk 2]
```diff
@@ -232,6 +232,25 @@ def test_aws_template_applies_labels_to_volume_tags() -> None:
...
+    WorkloadPriorityClass (the skypilot-priority-class annotation alone is
+    not read by Kueue)."""
+    template_path = pathlib.Path('sky/templates/kubernetes-ray.yml.j2')
+    template = template_path.read_text(encoding='utf-8')
+
+    expected_block = """            {% if k8s_kueue_local_queue_name %}
+            kueue.x-k8s.io/queue-name: {{k8s_kueue_local_queue_name}}
+            kueue.x-k8s.io/pod-group-name: {{cluster_name_on_cloud}}
+            {% if priority_class is not none %}
+            kueue.x-k8s.io/priority-class: {{priority_class}}
```
[the remark this answers is not in the document]
Comment (luccabb) [reply tests/unit_tests/test_backend_utils.py]: Done in e832338 (updated together with the template).
